com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -334,6 +361,8 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
             if (requiresDepthPrepass && this.actualRenderingMode == RenderingMode.Deferred && !renderPassInputs.requiresNormalsTexture)
                 requiresDepthPrepass = false;
 
+            requiresDepthPrepass |= m_DepthPrimingMode == DepthPrimingMode.Forced && m_CanUseDepthPriming;
```

Comment:
Maybe we should allow depth priming for "Forced" even on mobile GPUs with HSR.
There are probably some games where depth priming may still be beneficial in some situations.

Reply:
There are some situations where we could avoid the perf cost on Tiled Hardware. 
- When MSAA is not in use.
- When we can merge the depth prepass and opaque pass into one to avoid having to store/load MSAA data between passes.

If a GPU actually has HSR I'm not sure where we could gain perf. Wouldn't we be doing the same thing that already happens in the hardware ?  If we have  depth priming on (so depth write off and compare to equals) could that further speed up the HSR rendering ?
ARM has tiled hardware w/ automatic sorting so there we could gain the benefits of depth priming still IF we navigate the situations that we can actually use it.

Right now the plan is to add depth priming support for these situations in a later PR.

Reply:
At least on Mali and Adreno the HSR implementations cannot avoid all redundant fragment shader executions.
For example Adreno only tests against a low resolution depth buffer. There are also slow paths where it cannot do HSR at all. It's kind of a black box.
Based on heuristics the Adreno driver it may also switch to full immediate rendering without any binning or HSR.

Even PowerVR and similar will have to disable HSR if the fragment shaders writes to depth, for alpha-test or if there are other side-effects such as UAV writes.